src/data_access/JSONPersistence.java

```diff
@@ -0,0 +1,155 @@
+package data_access;
+
+import entity.Recipe;
+import entity.RecipeFactory;
+import entity.User;
+import entity.UserFactory;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+
+public class JSONPersistence implements Persistence {
+    private final UserFactory userFactory;
+    private final String filePath;
+    private final RecipeAPI recipeDAO;
+
+    public JSONPersistence(UserFactory userFactory, String filePath, RecipeAPI recipeDAO) {
+        this.userFactory = userFactory;
+        this.filePath = filePath;
+        this.recipeDAO = recipeDAO;
+    }
+
+    /**
+     * Make a JSONObject out of the tags for a given user.
+     * @param user the User object which we make the JSONObject based on.
+     * @return JSONObject containing a "tags" key and then an array of
+     * JSONObjects, each of which has a key of the tag name, with an associated
+     * JSONArray for the recipe IDs.
+     */
+    private JSONObject JSONForUserTags(User user) {
+        JSONObject obj = new JSONObject();
+        JSONArray array = new JSONArray();
```

Comment:
The JSONArray should be created within the loop for each tag to avoid accumulating recipe IDs for all tags in a single array and unintentionally duplicating recipes across multiple tags.